core/services/ocr2/plugins/ocr2keeper/evm21/mercury/v02/request.go

```diff
@@ -0,0 +1,175 @@
+				state = m.State
+			}
+			continue
+		}
+		results[m.Index] = m.Bytes[0]
+	}
+	// only retry when not all successful AND none are not retryable
+	return state, mercury.UpkeepFailureReasonNone, results, retryable && !allSuccess, reqErr
+}
+
+func (c *client) singleFeedRequest(ctx context.Context, ch chan<- mercury.MercuryData, index int, sl *mercury.StreamsLookup) {
+	q := url.Values{
+		sl.FeedParamKey: {sl.Feeds[index]},
+		sl.TimeParamKey: {sl.Time.String()},
+	}
+	mercuryURL := c.mercuryConfig.Credentials().GetLegacyURL()
+	reqUrl := fmt.Sprintf("%s%s%s", mercuryURL, mercuryPathV02, q.Encode())
+	c.lggr.Debugf("request URL for upkeep %s feed %s: %s", sl.UpkeepId.String(), sl.Feeds[index], reqUrl)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqUrl, nil)
+	if err != nil {
+		ch <- mercury.MercuryData{Index: index, Error: err, Retryable: false, State: mercury.InvalidMercuryRequest}
+		return
+	}
+
+	ts := time.Now().UTC().UnixMilli()
+	signature := mercury.GenerateHMACFn(http.MethodGet, mercuryPathV02+q.Encode(), []byte{}, c.mercuryConfig.Credentials().GetUsername(), c.mercuryConfig.Credentials().GetPassword(), ts)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", c.mercuryConfig.Credentials().GetUsername())
+	req.Header.Set("X-Authorization-Timestamp", strconv.FormatInt(ts, 10))
+	req.Header.Set("X-Authorization-Signature-SHA256", signature)
+
+	// in the case of multiple retries here, use the last attempt's data
+	state := mercury.NoPipelineError
+	retryable := false
+	sent := false
+	retryErr := retry.Do(
+		func() error {
+			retryable = false
+			resp, err := c.httpClient.Do(req)
+			if err != nil {
+				c.lggr.Warnf("at block %s upkeep %s GET request fails for feed %s: %v", sl.Time.String(), sl.UpkeepId.String(), sl.Feeds[index], err)
+				retryable = true
+				state = mercury.MercuryFlakyFailure
+				return err
+			}
+			defer resp.Body.Close()
+
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				state = mercury.InvalidMercuryResponse
+				return err
+			}
+
+			switch resp.StatusCode {
+			case http.StatusNotFound, http.StatusInternalServerError:
+				c.lggr.Warnf("at block %s upkeep %s received status code %d for feed %s", sl.Time.String(), sl.UpkeepId.String(), resp.StatusCode, sl.Feeds[index])
+				retryable = true
+				state = mercury.MercuryFlakyFailure
+				return errors.New(strconv.FormatInt(int64(resp.StatusCode), 10))
+			case http.StatusOK:
+				// continue
+			default:
+				state = mercury.InvalidMercuryRequest
+				return fmt.Errorf("at block %s upkeep %s received status code %d for feed %s", sl.Time.String(), sl.UpkeepId.String(), resp.StatusCode, sl.Feeds[index])
+			}
+
+			c.lggr.Debugf("at block %s upkeep %s received status code %d from mercury v0.2 with BODY=%s", sl.Time.String(), sl.UpkeepId.String(), resp.StatusCode, hexutil.Encode(body))
+
+			var m MercuryV02Response
+			if err := json.Unmarshal(body, &m); err != nil {
```

Comment:
since we are refactoring now, maybe let's use err1 here to avoid shadow?

Reply:
I prefer shadowing err over err1, err2 etc

Reply:
The golangci-lint configuration insists that you do not shadow. Is it failing to detect this case?

Reply:
@jmank88  I dont see lint issues. Can I run `golangci-lint` locally? if so, i can try to repro.

Reply:
This action is failing: https://github.com/smartcontractkit/chainlink/actions/runs/6791417403/job/18462870780#step:3:1

Reply:
There is make golangci-lint for the core module. Not sure about the others but it could be adapted easily